src/simulation/m_bubbles.fpp

```diff
@@ -281,34 +312,41 @@ contains
         !!  @param fV Current bubble velocity
         !!  @param fR0 Equilibrium bubble radius
         !!  @param fC Current sound speed
-    function f_rddot_KM(fpbdot, fCp, fCpbw, fRho, fR, fV, fR0, fC)
+        !!  @param gam_l Physical Bubble parameter (pass in from global parameters)
+        !!  @param Ca_l Cavitation number (pass in from global parameters)
+        !!  @param Web_l Weber number (pass in from global parameters)
+        !!  @param Re_inv_l Inverse Reynolds number (pass in from global parameters)
+        !!  @param polytropic_l Polytropic switch (pass in from global parameters)
+    pure elemental function f_rddot_KM(fpbdot, fCp, fCpbw, fRho, fR, fV, fR0, fC, gam_l, Ca_l, Web_l, Re_inv_l, polytropic_l)
         !$acc routine seq
         real(wp), intent(in) :: fpbdot, fCp, fCpbw
         real(wp), intent(in) :: fRho, fR, fV, fR0, fC
+        real(wp), intent(in) :: gam_l, Ca_l, Web_l, Re_inv_l
+        logical, intent(in) :: polytropic_l
```

Comment:
Similar to above, since these are `intent(in)` why was the refactor necessary to make it `pure`?